test/distributed/pipelining/test_pipe.py

```diff
@@ -22,19 +22,18 @@
 class ExampleCode(torch.nn.Module):
     def __init__(self):
         super().__init__()
-        self.mm_param0 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
         self.mm_param1 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
         self.mm_param2 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
         self.lin1 = torch.nn.Linear(d_hid, d_hid)
         self.lin2 = torch.nn.Linear(d_hid, d_hid)
 
     def forward(self, x, y):
-        x = torch.mm(x, self.mm_param0)
+        x = torch.mm(x, self.mm_param1)  # mutli-use param
         skip_connection = x
         x = x + y
         x = torch.relu(x)
         pipe_split()
-        x = torch.mm(x, self.mm_param1)
+        x = torch.mm(x, self.mm_param1)  # mutli-use param
```

Comment:
do we have tests that verify fqn sanity (perhaps you added them along with unflattener)?

it'd be nice to confirm that when using multi-use param, the model's state_dict is clean and only has the original copy so checkpoint save/load will work as expected.

Reply:
tbh, we don't have support for multi-use param in training yet. Because that would require an all-reduce between the multiple copies of that param, because the next batch forward happens. So, it would be kind of early to talk about how to save them before we can train them :)

Reply:
But, multi-use buffer (as in titan case) and multi-user param in inference are different stories, they can be supported today.